include/api/CSingleFieldDataCategorizer.h

```diff
@@ -105,6 +118,11 @@ class API_EXPORT CSingleFieldDataCategorizer {
                                       const CCategoryIdMapper& categoryIdMapper,
                                       core::CStatePersistInserter& inserter);
 
+    //! Write the latest categorizer stats, plus an annotation if the
+    //! categorization status has changed.
+    void writeStats(CJsonOutputWriter& jsonOutputWriter,
+                    CAnnotationJsonWriter& annotationJsonWriter);
```

Comment:
I feel since you now have `writeStatsIfUrgent` it would be more consistent to call this `writeStatsIfStatusChanged`.

Reply:
It's called from two places, and writes the stats if _any_ field within the stats has changed.  I'll change it to `writeStatsIfChanged`.